api/v1alpha1/servicedefaults_types.go

```diff
@@ -153,12 +240,20 @@ func (in *ServiceDefaults) Validate(namespacesEnabled bool) error {
 	var allErrs field.ErrorList
```

Comment:
should `ToConsul` function convert UpstreamConfig?

Reply:
Maybe we should also add a test that sets all possible values set for service defaults and verifies that it gets written to consul.

Reply:
Yeah..this was definitely a massive oversight by me